### PR TITLE
Math optimization

### PR DESCRIPTION
**Pull request by ElisaBortoli**
**[WIP]**
PR in which a faster power function is defined:
```
template <class T>
T power(const T a, const T b){
#ifdef MATH_OPTIM
    return std::exp(b*std::log(a));
#else
    return std::pow(a,b);
#endif
}
```

I've tried to put it in 
- the function **atm_transmittance**
- the parameter ```psisat``` in **pedo.func.cc**
BUT
even if the short tests, both 1D and 3D, are all passing, with longer tests the output values differ from the 2.0 of up to e+1 in absolute values and e+0 in relative values.

I tabulated values of bases and exponents for the functions in the **atm_transmittance** and of ```psisat``` but they're not negative so the sobstitution should work. Some errors appear for very small and very high values of the basis but I'm not sure I can justified sunch high differences.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ before_install:
   - export PATH="$PWD/_meson:$PATH"
 
 install:
-  - meson.py -DMUTE_GEOLOG=true -DMUTE_GEOTIMER=true build
+  - meson.py -DMUTE_GEOLOG=true -DMUTE_GEOTIMER=true -DMATH_OPTIM=true build
```
Comment (asartori86) [inline .travis.yml]: remove `-DMATH_OPTIM=true` and we are done

File: tests/1D/Matsch_B2_Ref_007/geotop.inpts [hunk 2]
```diff
@@ -8,7 +8,7 @@
  
 TimeStepEnergyAndWater	=	900
 InitDateDDMMYYYYhhmm	=	02/10/2009 00:00  
-EndDateDDMMYYYYhhmm	=	02/11/2009 00:00
+EndDateDDMMYYYYhhmm	=	31/12/2015 23:00
```
Comment (asartori86) [inline tests/1D/Matsch_B2_Ref_007/geotop.inpts]: how long this test takes on travis?

File: .travis.yml (hunk 1 above)
Comment (ElisaBortoli) [reply .travis.yml]: done

File: src/geotop/math.optim.h [hunk 3]
```diff
@@ -1,17 +1,38 @@
...
 #include <cmath>
+#include <fstream>
 
 #define pow_2(a) ((a)*(a))
 
 template <class T>
 T power(const T a, const T b){
 #ifdef MATH_OPTIM
-  return std::exp(b*std::log(a));
+    /* std::ofstream myfile{"analyze_power.txt", std::ios_base::app}; */
```
Comment (asartori86) [inline src/geotop/math.optim.h]: please remove the comments

File: src/geotop/math.optim.h [hunk 4]
```diff
@@ -1,17 +1,38 @@
...
+    /* file << "(" << a << "," << b << ")\t\t" << std::endl; */
+    return std::pow(a,b);
 #endif
 }
 
-#endif 
+#endif
+
+
+//template <class T>
```
Comment (asartori86) [inline src/geotop/math.optim.h]: and here

File: src/geotop/math.optim.h (hunk 3 above)
Comment (ElisaBortoli) [reply src/geotop/math.optim.h]: done

File: src/geotop/math.optim.h (hunk 4 above)
Comment (ElisaBortoli) [reply src/geotop/math.optim.h]: done

File: src/geotop/math.optim.h [hunk 5]
```diff
@@ -1,17 +1,20 @@
 #ifndef __math_optim_h_
 #define __math_optim_h_
 
+#include "config.h"
+
 #include <cmath>
+#include <fstream>
```
Comment (asartori86) [inline src/geotop/math.optim.h]: why you need this header here?

Comment (ElisaBortoli) [reply src/geotop/math.optim.h]: it was useful to print the bases and the exponents to a file but now it has to be removed

**Pull request merged**